### PR TITLE
Medical Treatment - Add preferred seats for loading patients

### PR DESCRIPTION
**Pull request by BrettMayson**
**When merged this pull request will:**
- Defines seats to prioritize when loading patients into vehicles
- Documentation for mod makers to add it to their vehicles
- If there are no preferred seats available, patients are loaded from highest cargo seat to lowest

https://user-images.githubusercontent.com/3946739/150858689-959180fa-bce7-4bb0-9b01-aa33ec19fb07.mov

### CODE REVIEW

**Comment by jonpas** [comment]
I think the default patient/captive load order should start from the last passenger anyways (technically last, as that covers most cases and prevents loading into co-driver seat). It's always super annoying to load a patient/captive and he is in the front (first passenger seat).

Additional config for it is nice to tweak it further. :+1:

**Comment by BrettMayson** [comment]
It'd be really nice to have a command to get the available cargo seats. *cough cough*

**Comment by jonpas** [comment]
> It'd be really nice to have a command to get the available cargo seats.

How does ACE Interaction for seats work? It even distinguishes Passenger seats. I would use same logic, and prioritize with reverse indices.

**Comment by BrettMayson** [comment]
> It'd be really nice to have a command to get the available cargo seats. cough cough
Ah it's `fullCrew`, it's not grouped with the cargo commands on the biki

File: addons/medical_treatment/functions/fnc_loadUnit.sqf [hunk 1]
```diff
@@ -32,7 +32,13 @@ if (_patient call EFUNC(medical_status,isBeingDragged)) then {
...
 };
 
-private _vehicle = [_medic, _patient, _vehicle] call EFUNC(common,loadPerson);
+private _preferredSeats = if (!isNull _vehicle) then {
+    getArray (configOf _vehicle >> QGVAR(patientSeats))
+} else {
+    []
+};
+
+private _vehicle = [_medic, _patient, _vehicle, _preferredSeats, true] call EFUNC(common,loadPerson);
```
Comment (veteran29) [inline addons/medical_treatment/functions/fnc_loadUnit.sqf:41]: Could you get "reverse fill" from the config too?
```sqf
([configOf _vehicle >> QGVAR(reverseFill), "NUMBER", 1] call CBA_fnc_getConfigEntry) > 0
```

File: addons/medical_treatment/functions/fnc_loadUnit.sqf [hunk 2]
```diff
@@ -32,7 +32,19 @@ if (_patient call EFUNC(medical_status,isBeingDragged)) then {
...
+    []
+};
+
+private _vehicle = [
+    _medic,
+    _patient,
+    _vehicle,
+    _preferredSeats,
+    ([configOf _vehicle >> QGVAR(patientReverseFill), "NUMBER", 1] call CBA_fnc_getConfigEntry) > 0
+] call EFUNC(common,loadPerson);
```
Comment (veteran29) [inline addons/medical_treatment/functions/fnc_loadUnit.sqf:47]: there's no need for null check:
```sqf
_configNull = configOf objNull;
getArray (_configNull >> "asd") // returns []
```

```suggestion
private _vehicle = [
    _medic,
    _patient,
    _vehicle,
    getArray (configOf _vehicle >> QGVAR(patientSeats)),
    ([configOf _vehicle >> QGVAR(patientReverseFill), "NUMBER", 1] call CBA_fnc_getConfigEntry) > 0
] call EFUNC(common,loadPerson);
```

**Review by veteran29 — APPROVED** [review]

File: docs/wiki/framework/medical-treatment-framework.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  major: 3
+  minor: 14
+  patch: 2
+---
+
+## 1. Config Values
+
+### 1.1 Vehicle Patient Seats
+
+Defines the seats that will be prioritized when loading patients into vehicles.
```
Comment (PabstMirror) [inline docs/wiki/framework/medical-treatment-framework.md:19]:
```suggestion
Defines the seats that will be prioritized when loading patients into vehicles. Uses `moveInCargo` indexes.
```
Not really sure if this is needed, but there are sorta two different numbering systems. The other being for https://community.bistudio.com/wiki/Arma_3:_Actions#GetInCargo

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**